src/iop/colorin.c

```diff
@@ -932,7 +932,7 @@ static void process_cmatrix_fastpath(struct dt_iop_module_t *self,
   // figure out the number of pixels each thread needs to process
   // round up to a multiple of 4 pixels so that each chunk starts aligned(64)
   const size_t nthreads = dt_get_num_threads();
-  const size_t chunksize = 4 * (((npixels / nthreads) + 3) / 4);
+  const size_t chunksize = 4 * (((npixels / nthreads) + 4) / 4);
```

Comment:
On working through this, I see that the problem is that the original version computes `floor(npixels/nthreads)` while we need `ceil(npixels/nthreads)`.  So the actual correct computation with minimal disparity in work between threads is
```
4 * ((((npixels + nthreads - 1) / nthreads) + 3) / 4)
```
Your proposal is equivalent to (npixels+nthreads) instead of (npixels+nthreads-1), so either is fine.